mget.go

```diff
@@ -0,0 +1,128 @@
+package opensearchtools
+
+import (
+    "bytes"
+    "context"
+    "encoding/json"
+    "net/http"
+
+    "github.com/opensearch-project/opensearch-go/v2"
+    "github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+)
+
+// MGetRequest wraps the functionality of [opensearchapi.MgetRequest] by supporting request body creation.
+// This is done with a generic K [RoutableDoc] letting you [Add] documents to be requested.
+// For example, if we use the BasicDoc example from [RoutableDoc] we can leverage an MGetRequest as simply as:
+//
+//	doc := BasicDoc{ID: "sampleId", Index: "index"}
+//	mgetResults, mgetError := NewMGetRequest[BasicDoc]().Add(doc).Do(context.background(), client)
+type MGetRequest[K RoutableDoc] struct {
+    // Index destination for entire request
+    // if used individual documents don't need to specify the index
+    Index string `json:"-"`
+
+    Docs []mgetDoc `json:"docs"`
+}
+
+// NewMGetRequest instantiates an empty MGetRequest for declared [RoutableDoc] K.
+// An empty MGetRequest is executable but will return zero documents because zero have been requested.
+func NewMGetRequest[K RoutableDoc]() *MGetRequest[K] {
+    return &MGetRequest[K]{}
+}
+
+// Add any number of docs to the MGetRequest
+func (m *MGetRequest[K]) Add(docs ...K) {
```

Comment:
@cs-marshall does it make sense to have a few other methods here like AddWithIndexAndId(index, id int), etc?
Seems like sometimes one might want to do without defining a type which implements K... WDYT?

Reply:
Maybe make this `Add(index, id string)` and add a separate `AddDocs(docs ...K)`?

Reply:
> without defining a type which implements K

have to do this regardless to use `MGetRequest` at all because of the generic constraint.

Reply:
`Add` using the docs is a bit of a copy pasta from olivere mget, but I do think `AddDocs` is worth keeping. I've got no problem having both methods though.

Reply:
maybe best to leave this as is, since `K` has to be a `RoutableDoc` anyway and consumers can just use the OS client directly to mget using only id/index

Reply:
That being said, I think it would be good UX to add a package-level `MGet[K RoutableDoc](ctx context.Context, client *opensearch.Client, docs ...K) (MGetResponse[K], error)` function that wraps creating an `MGetRequest` and calling `Do()` on it.

Reply:
I'm on the fence about the `Add` utility functions. I can think of three core ones
```
// Add a basic fetch request. The id must return a document of type K when fetched or the request will fail
func (m *MGetRequest[K]) Add(id, index string) 

// AddDocs to be fetched by the MultiGet Request
func (m *MGetRequest[K]) AddDocs(docs...K)

// AddIds to be fetch. This relies on the MGetRequest having a top level index assigned to use for all IDs.
// The fetched documents must be of type K when fetched, or the request will fail.
func (m *MGetRequest[K]) AddIds(id...string)
```

Though writing that out, that does have the type safety issue of the ids not necessarily leading to a K document. Though that can happen even if someone uses a RoutableDoc implementation.

Reply:
As for the utility function...

```
func MGet[K RoutableDoc](ctx context.Context, client *opensearch.Client, docs ...K) (*MGetResponse[K], error) {
	return NewMGetRequest[K]().Add(docs...).Do(ctx, client)
}
```

It's less typing, but idk if it's that useful. The wrapped code isn't that much either... I'll add it if you still like it, but idk if I'm convinced.

Reply:
It would change the consumer usage from
```
req := opensearchtools.NewMGetRequest[SomeDoc]()
req.Add(docs...)
res, err := req.Do(ctx, client)
```
to
```
res, err := opensearchtools.MGet(ctx, client, docs...)
```
I know the API is technically fluent and the first snippet _could be_ done on 1 line, but that pattern is generally considered un-idiomatic.  The function version also takes advantage of type inference to avoid having to explicitly type the `[SomeDoc]`.

Reply:
ooohhhh type inference, completely forgot about that.

Why is that pattern of builder un-idiomatic in go? I really like the pattern...

Reply:
Fluent APis in general are uncommon and unpopular in my experience, probably because they have a tendency to get clunky when you add in error handling.

Reply:
The trick is not to have errors 😉. I'm sure it's a holdover from my java days, but I like the feel of fluent apis when the struct can be complex and have various structs as attributes, rather than the literal construction in GO.

Reply:
Add the package function and we can do both. /js